File: Domain/Entities/Customer.cs
namespace Domain.Entities;

public class Customer
{
    public int Id { get; set; }

    public string Name { get; set; }

    public string? Phone { get; set; }

    public string? IdentityNumber { get; set; }

    public string? RoomNumber { get; set; }

    public ICollection<Order> Orders { get; set; } = new List<Order>();
}

File: app/Models/Customer.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;

class Customer extends Model
{
    protected $fillable = [
        'name',
        'phone',
        'identity_number',
        'room_number',
        'notes',
    ];


    public function operations(): HasMany
    {
        return $this->hasMany(Operation::class);
    }


    public function transactions(): HasMany
    {
        return $this->hasMany(Transaction::class);
    }
}
